feat: add overlay-recommended-packages (#1651)
canonical/craft-parts: docs/common/craft-parts/explanation/overlay_parameters.rst

Overlay parameters
------------------

A part has five parameters that can be used to adjust how the overlay step
works: ``overlay-packages``, ``overlay-recommended-packages``, ``overlay-script``,
``override-overlay`` and ``overlay``. ``overlay-packages`` and ``overlay`` (the
overlay-files parameter) behave much the same way as the related parameters on the
``STAGE`` step. The ``overlay-recommended-packages`` key works like

> ``overlay-packages``, but installs packages with their recommended packages.

Reviewer: ```suggestion
``overlay-packages``, but installs packages with their recommended dependencies.
```

> ``overlay-packages``, but installs packages with their recommended dependencies.

Reviewer 2: What do you think about linking to https://www.debian.org/doc/manuals/debian-faq/pkg-basics.en.html#depends ?

> ``overlay-packages``, but installs packages with their recommended dependencies.

Reviewer 2: Since `Recommends` is separate from `Depends`:

```suggestion
``overlay-packages``, but also installs any `recommended packages
<https://www.debian.org/doc/manuals/debian-faq/pkg-basics.en.html#depends>`__ that
the target packages might have.
```

The ``overlay-script`` and ``override-overlay`` keys both behave

> similarly to ``override-stage`` and are mutually incompatible.

Reviewer 2: To avoid ambiguity:

```suggestion
The ``overlay-script`` and ``override-overlay`` keys both take as value a scriptlet 
that replaces the default overlay step behavior. They are mutually incompatible.
```

Overlay parameters
------------------

A part has five parameters that can be used to adjust how the overlay step
works: ``overlay-packages``, ``overlay-recommended-packages``, ``overlay-script``,
``override-overlay`` and ``overlay``. ``overlay-packages`` and ``overlay`` (the
overlay-files parameter) behave much the same way as the related parameters on the
``STAGE`` step. The ``overlay-recommended-packages`` key works like
``overlay-packages``, but also installs any `recommended packages

> <https://www.debian.org/doc/manuals/debian-faq/pkg-basics.en.html#depends`__ that the

Reviewer 2: ```suggestion
<https://www.debian.org/doc/manuals/debian-faq/pkg-basics.en.html#depends>`__ that the
```
